src/client/sync-client.ts

```diff
@@ -182,6 +210,94 @@ export function makeSyncClient(opts: SyncClientOptions = {}): SyncClient {
       }
 
       throw error
+    },
+
+    async syncRepo(disklet, syncKey, lastHash) {
```

Comment:
This function seems reasonable. However, this should not be public API. Instead, we want to export a `makeSyncedDisklet(baseDisklet: Disklet) => Disklet & { sync() => Promise<void> }`. You do `const myDisklet = makeSyncedDisklet(...)`, and then you can do `myDisklet.getText`, `myDisklet.delete`, `myDisklet.setText`, and so forth, and the changes go into the right places (/data, /changes, /deleted). Finally, you can do `myDisklet.sync()` to sync with the server changes. The contents of `syncRepo` are essentially the `sync` implementation - great! What's missing is the rest of the algorithm.